event_routing_backends/helpers.py

```diff
@@ -93,17 +93,16 @@ def get_user(username_or_id):
     Returns:
         user object
     """
-    user = user_id = username = None
-    if username_or_id:
-        try:
-            user_id = int(username_or_id)
-        except ValueError:
-            username = username_or_id
+    user = username = None
+
+    if not username_or_id:
+        return None
 
-    if username:
+    try:
+        user = User.objects.get(id=int(username_or_id))
```

Comment:
What if the username is equal to a user ID in the database?

Reply:
the first thing, that current implementation makes, is to try to set  the `user_id` [here](https://github.com/openedx/event-routing-backends/blob/master/event_routing_backends/helpers.py#L99) and then look for the user with that id , so if there is a user with `username=5` and another user with `id=5`, this will return the user with `id=5`, and  this basically keeps that logic. The modification implemented here considers the case when there is no a user with `id=5` instead of returning None this will return the user with `username =5`. 

Finally my comment is that there is not an easy way  to identify the right case, the method uses  `username_or_id` parameter but that is too ambiguous  and the method can not identify that by itself that needs some criteria like first id and then username or limit the method scope to username or id, not both, but that requires to standard the events to use just one of them.

Reply:
What if we verify if the user_id is already an integer (all events send this param as int) then we query by ID, in another case query by username?